src/Setup/UpdateVersions.purs

```diff
@@ -60,115 +71,131 @@ updateVersions = do
 -- | releases, falls back to the highest valid semantic version tag for the tool.
 fetchLatestReleaseVersion :: Tool -> Aff Version
 fetchLatestReleaseVersion tool = Tool.repository tool # case tool of
-  PureScript -> fetchFromGitHubReleases
-  Spago -> fetchFromGitHubReleases
+  PureScript -> fetchFromGitHubReleases OnlyRealReleases
+  Spago -> fetchFromGitHubReleases OnlyRealReleases
   Psa -> fetchFromGitHubTags
   PursTidy -> fetchFromGitHubTags
-  Zephyr -> fetchFromGitHubReleases
-  where
-  -- TODO: These functions really ought to be in ExceptT to avoid all the
-  -- nested branches.
-  fetchFromGitHubReleases repo = recover do
-    page <- liftEffect (Ref.new 1)
-    untilJust do
-      versions <- liftEffect (Ref.read page) >>= toolVersions repo
-      case versions of
-        Just versions' -> do
-          let version = Array.find (not <<< Version.isPreRelease) versions'
-          when (isNothing version) do
-            liftEffect $ void $ Ref.modify (_ + 1) page
-          pure version
-
-        Nothing ->
-          throwError $ error "Could not find version that is not a pre-release version"
-
-  toolVersions :: Tool.ToolRepository -> Int -> Aff (Maybe (Array Version))
-  toolVersions repo page = do
-    let
-      url =
-        "https://api.github.com/repos/"
-          <> repo.owner
-          <> "/"
-          <> repo.name
-          <> "/releases?per_page=10&page="
-          <> show page
-
-    AX.get RF.json url >>= case _ of
-      Left err -> throwError (error $ AX.printError err)
-      Right { body } -> case decodeJson body of
-        Left e -> do
-          throwError $ error
-            $ fold
+  Zephyr -> fetchFromGitHubReleases OnlyRealReleases
+
+-- TODO: These functions really ought to be in ExceptT to avoid all the
+-- nested branches.
+fetchFromGitHubReleases :: ReleaseType -> Tool.ToolRepository -> Aff Version
+fetchFromGitHubReleases releaseType repo = recover do
+  page <- liftEffect (Ref.new 1)
+  let
+    releaseFilter = case releaseType of
+      OnlyRealReleases ->
+        not <<< Version.isPreRelease
+      OnlyPreReleases expectedV -> do
+        let
+          equating f a b = (f a) == (f b)
+          majorMinorMatch v = equating Version.major expectedV v && equating Version.minor expectedV v
+        majorMinorMatch && Version.isPreRelease
```

Comment:
Ideally, `majorMinorMatch` would include the `patch`, but `purescript-strings` doesn't export that function.

Reply:
We can use `Version.patch`, no?